typing/typedecl.ml

```diff
@@ -915,13 +903,14 @@ let transl_declaration env sdecl (id, uid) =
             env None true lbls (Record { unboxed = unbox })
           in
           let rep, jkind =
+            if unbox then
+              Record_unboxed,
+              Jkind.of_new_legacy_sort ~why:Old_style_unboxed_type
+            else
```

Comment:
I agree that in practice we were limited to value here, so we're not losing anything.  But it does seem a bit sad to restrict this to be representable - it goes against our general goal to have layouts be restricted by uses rather than here in typedecl.

I don't think we should hold up the PR for it, but can you say more about what inference problem leaving this as `any` causes?

Reply:
No, sorry, I can't. I don't seem to remember. But I do remember not investing time into understanding this better, thinking that we would fix it in the sweep to delay checks for `any` to construction sites.